src/components/AnalyticsFilterBar/AnalyticsFilterBar.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { ComponentProps, useCallback, useMemo, useState } from 'react';
+import { HireStream } from '@prisma/client';
+import { nullable } from '@taskany/bricks';
+import {
+    Button,
+    FiltersBar,
+    FiltersBarItem,
+    FiltersBarControlGroup,
+    FiltersBarTitle,
+    Separator,
+} from '@taskany/bricks/harmony';
+
+import { useAnalyticsFilterUrlParams } from '../../hooks/useAnalyticsFilterUrlParams';
+import { AddFilterDropdown } from '../AddFilterDropdown';
+import { AppliedAnalyticsPeriodFilter } from '../AppliedAnalyticsPeriodFilter/AppliedAnalyticsPeriodFilter';
+import { AppliedAnalyticsStreamsFilter } from '../AppliedAnalyticsStreamsFilter/AppliedAnalyticsStreamsFilter';
+
+import { tr } from './AnalyticsFilterBar.i18n';
+import s from './AnalyticsFilterBar.module.css';
+
+interface AnalyticsFilterBarProps {
+    title?: string;
+    hireStreams?: HireStream[];
+}
+
+export const AnalyticsFilterBar = ({ title, hireStreams }: AnalyticsFilterBarProps): JSX.Element => {
+    const { values, setter, clearParams, setPeriod, isFiltersEmpty } = useAnalyticsFilterUrlParams(hireStreams);
+
+    const [showStreamsDropdown, setShowStreamsDropdown] = useState(Boolean(values.streams));
+    const [showPeriodDropdown, setShowPeriodDropdown] = useState(Boolean(values.period));
+
+    const unusedFilterItems = useMemo(() => {
+        const items = [];
+        if (!values.streams && !showStreamsDropdown && hireStreams) items.push({ id: 'streams', title: tr('Streams') });
+        if (!values.period && !showPeriodDropdown) items.push({ id: 'period', title: tr('Period') });
+        return items;
+    }, [values, showStreamsDropdown, showPeriodDropdown, hireStreams]);
+
+    const resetFilters = useCallback(() => {
+        setShowStreamsDropdown(false);
+        setShowStreamsDropdown(false);
```

Comment:
really needs to call function many times?

Reply:
fixed